cfme/configure/configuration/server_settings.py

```diff
@@ -219,6 +218,78 @@ def basic_information_values(self):
         view = navigate_to(self, 'Details')
         return view.basic_information.read()
 
+# =============================== Server Roles Form ===================================
+
+    def update_server_roles_ui(self, updates, reset=False):
+        """ Navigate to a Server Tab. Updates server roles via UI
+
+        Args:
+             updates: dict, widgets will be updated regarding updates.
+             reset: By default(False) changes will not be reset, if True changes will be reset
+
+        """
+        view = navigate_to(self, 'Details')
+        # embedded_ansible server role available from 5.8 version
+        if self.appliance.version < '5.8' and 'embedded_ansible' in updates:
+            updates.pop('embedded_ansible')
+        # cockpit_ws element is not present for downstream version
+        if self.appliance.version != self.appliance.version.latest() and 'cockpit_ws' in updates:
+            updates.pop('cockpit_ws')
+        view.server_roles.fill(updates)
+        self._save_action(view, updates, reset)
+
+    def update_server_roles_db(self, roles):
+        """ Set server roles on Configure / Configuration pages.
+
+        Args:
+            roles: Roles specified as in server_roles dict in this module. Set to True or False
+        """
+        if self.server_roles_db == roles:
+            logger.debug(' Roles already match, returning...')
+            return
+        else:
+            self.appliance.server_roles = roles
+
+    @property
+    def server_roles_db(self):
+        """ Get server roles from Configure / Configuration from DB
+
+        Returns: :py:class:`dict` with the roles in the same format as :py:func:`set_server_roles`
+            accepts as kwargs.
+        """
+        return self.appliance.server_roles
+
+    @property
+    def server_roles_ui(self):
+        view = navigate_to(self, 'Details')
+        roles = view.server_roles.read()
+        # default_smart_proxy is not a role, but text info
+        roles.pop('default_smart_proxy')
+        return roles
+
+    def server_roles_enabled(self, *roles):
```

Comment:
REQUIRED: Since this is not used as a context manager anymore, it should be renamed to `enable_server_roles`. Same with `server_roles_disabled` below - it should be `disable_server_roles`.